rosetta/retriever/retriever.go

```diff
@@ -350,12 +352,20 @@ func (r *Retriever) operations(txID flow.Identifier, events []flow.Event) ([]*ob
 	// Now we can convert each event to an operation, as they are both filtered for
 	// only supported ones and properly ordered.
 	ops := make([]*object.Operation, 0, len(filtered))
-	for index, event := range filtered {
+
+	// Keep track of event indices while ignoring skipped events.
+	index := 0
```

Comment:
Wouldn't a cleaner approach just filter out events that have an empty address before looping?